integrations/utils/request.js

```diff
@@ -0,0 +1,38 @@
+import { HTTPError, NetworkConnectivityError } from './errors';
+
+/**
+ *  A wrapper around fetch which handles errors and JSON nicely
+ * @param {String} url    The url to fetch data from
+ * @param {Object} opts   Fetch options, including data
+ * @return {Object}       response object
+ */
+export default async function request(url, opts = {}) {
```

Comment:
Maybe we should call this `requestJSON`?
I would also prefer a signature that is more explicit with a `headers` and `body` (or `data`) argument

Reply:
Sure, have made it more explicit